### PR TITLE
Change update checker version source

### PR DESCRIPTION
**Pull request by KoBeWi**
https://github.com/godotengine/godot/pull/75916#discussion_r1566912471
Don't merge until the page is actually up :D

### CODE REVIEW

File: doc/classes/EditorSettings.xml [hunk 1]
```diff
@@ -875,7 +875,7 @@
 			All update modes will ignore builds with different major versions (e.g. Godot 4 -&gt; Godot 5).
 		</member>
 		<member name="network/connection/network_mode" type="int" setter="" getter="">
-			Determines whether online features are enabled in the editor, such as the Asset Library or update checks. Disabling these online features helps alleviate privacy concerns by preventing the editor from making HTTP requests to the Godot website, GitHub, or third-party platforms hosting assets from the Asset Library.
+			Determines whether online features are enabled in the editor, such as the Asset Library or update checks. Disabling these online features helps alleviate privacy concerns by preventing the editor from making HTTP requests to the Godot website, or third-party platforms hosting assets from the Asset Library.
```
Comment (AThousandShips) [inline doc/classes/EditorSettings.xml:878]:
```suggestion
			Determines whether online features are enabled in the editor, such as the Asset Library or update checks. Disabling these online features helps alleviate privacy concerns by preventing the editor from making HTTP requests to the Godot website or third-party platforms hosting assets from the Asset Library.
```

**Comment by coppolaemilio** [comment]
The page https://godotengine.org/versions.yml should serve the file needed. @KoBeWi please test it out to see if there is anything wrong/missing.

**Comment by akien-mga** [comment]
It's a bit more work on the website side, but shouldn't we expose the data as an easier-to-parse JSON so we can simplify the logic (and third-party project managers / hubs could also use it via our JSON parser, instead of manually implementing a YAML one)?

**Comment by coppolaemilio** [comment]
I'm fine with either, I think json is more web-friendly, so we should probably use that. Happy to make the adjustments needed, but would like to get at least specifications to generate the file with all the information required.

**Comment by KoBeWi** [comment]
The file loads fine.

We could replace it with JSON, but *technically* parsing the file line-by-line is more efficient (with JSON you need to parse the whole file and then search for data). JSON would make the code simpler, but since it's already written, it's fine either way.

As for the specs, it would be best if it was an array of dictionaries, sorted from newest to oldest. Each dictionary needs a version number and flavor/name. The download archive link can be generated from version and release notes are unused, so they don't need to be included.

**Comment by akien-mga** [comment]
> The download archive link can be generated from version and release notes are unused, so they don't need to be included.

I think the release notes could be used in the future, or by other users of this file, to link directly to release notes. So I'd suggest including them.

As for the download, I'd suggest including it too, in case we ever change the URLs for it.
Currently it's constructed as `"https://godotengine.org/download/archive/" + found_version`, but if we change this in the future, having it in the JSON means we can just update the JSON and all past Godot versions will still have working links instead of 404s.

Alternatively, we could include just the download archive URL "https://godotengine.org/download/archive/" as a field in the JSON, that can be read by any user and used to compose the full URL for each version, provided we're committed to always have such URLs like "URL_BASE/4.3-dev5".

**Comment by GreatNovaDragon** [comment]
I would muse, for further Future Proofing, it would be nice if the actual URL checked is an editor setting, kinda like how you can determine where to download the export templates in that URL?

**Comment by coppolaemilio** [comment]
I made the simplest conversion and the file is now a json file: https://godotengine.org/versions.json
If any other field should be included it can be added to the regular `_data/versions.yml`.

Let me know if further changes are needed!

**Comment by akien-mga** [comment]
> I made the simplest conversion and the file is now a json file: https://godotengine.org/versions.json If any other field should be included it can be added to the regular `_data/versions.yml`.
> 
> Let me know if further changes are needed!

Looks good! For ease of use, should we add the `https://godotengine.org` prefix to the release notes fields?

**Comment by coppolaemilio** [comment]
They all include the full path in the json file now 👍

**Comment by KoBeWi** [comment]
For some reason Godot loads only half of that file 🤔I printed the received string and it cuts abruptly.

**Comment by KoBeWi** [comment]
Updated to use the new JSON.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**